src/tooling/piral-cli/src/apps/new-pilet.test.ts

```diff
@@ -49,14 +52,21 @@ describe('New Pilet Command', () => {
     };
 
     // Act
-    const result = await newPilet(dir, options);
+    let result
+    try {
+      // In node14, this resolves to undefined.
+      // In node16, this rejects with an Error.
+      result = await newPilet(dir, options);
```

Comment:
Why does it reject? I think the test case should be left as is (unless there is something wrong with the expectation here) and the issue (`newPilet` not following expectations) should be fixed. What is thrown here?

Reply:
This is the tricky one where some NPM behavior (child process that we execute as part of the `new-pilet` tests) seems to have changed with the new version of NPM.  I haven't dug deeper into what the child process returns (e.g., whether the error code is different or whether there's different output to stdout/stderr).  What was happening with node16/npm7 is that it results in a rejected promise and will prematurely terminate the test without the try/catch wrapper around the call.

It might be worth looking and possibly refactoring the `scripts.ts` itself in the `piral-cli` code to handle the child process result in a more resilient fashion and abstract away the error returned from the NPM process execution.

Reply:
Yeah I'd prefer scripts.ts to normalize that behavior - such that the test remains unchanged.